spring-reactive-modules/spring-5-reactive-client-2/src/main/java/com/baeldung/limitrequests/client/DelayElements.java

```diff
@@ -0,0 +1,35 @@
+package com.baeldung.limitrequests.client;
+
+import java.time.Duration;
+
+import org.springframework.web.reactive.function.client.WebClient;
+
+import com.baeldung.limitrequests.client.utils.Client;
+import com.baeldung.limitrequests.client.utils.RandomConsumer;
+
+import reactor.core.publisher.Flux;
+
+public class DelayElements {
+
+    private DelayElements() {
+    }
+
+    public static Flux<Integer> fetch(WebClient client, int requests, int delay) {
+        String clientId = Client.id(requests, DelayElements.class, delay);
+
+        return Flux.range(1, requests)
+            .log()
+            .delayElements(Duration.ofMillis(delay))
+            .flatMap(i -> {
+                return RandomConsumer.get(client, clientId);
+            });
```

Comment:
This statement can be simplified with a expression lambda:

`            .flatMap(i -> RandomConsumer.get(client, clientId));`